test/jdk/javax/swing/JFileChooser/AquaDefaultButtonTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+/*
+ * @test
+ * @bug 8300084
+ * @key headful
+ * @summary Test to check if getDefaultButton() return valid default approve button.
+ * @requires (os.family == "mac")
+ * @run main AquaDefaultButtonTest
+ */
+
+public class AquaDefaultButtonTest {
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("com.apple.laf.AquaLookAndFeel");
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                JFileChooser fc = new JFileChooser();
+                JButton defApproveBtn = fc.getUI().getDefaultButton(fc);
+                if (defApproveBtn == null) {
```

Comment:
```suggestion
                JButton approve = fc.getUI().getDefaultButton(fc);
                if (approve == null) {
```
On the scope of two lines, its type — `JButton` — as well as method name makes it obvious it's a _button_.

`approveButton` would also be good.

Yet I am for removing this test completely because it's functionality is entirely covered by the `CustomApproveButtonTest.java` in #11901.